augur/tasks/git/dependency_tasks/core.py

```diff
@@ -16,52 +20,133 @@ def generate_deps_data(session, repo_id, path):
+        :param repo_id: Repository ID
+        :param path: URL path of the Repostiory
+    """
+    session.logger.info('Generating scorecard data for repo')
+    session.logger.info(f"Repo ID: {repo_id}, Path: {path}")
+
+    # we convert relative path in the format required by scorecard like github.com/chaoss/augur
+    # raw_path,_ = path.split('-')
+    # scorecard_repo_path = raw_path[2:]
+    path = path[8:]
+    if path[-4:] == '.git':
+        path = path.replace(".git", "")
+    command = '--repo='+ path
+    
+    #this is path where our scorecard project is located
+    path_to_scorecard = os.environ['HOME'] + '/scorecard'
+
+    #setting the environmental variable which is required by scorecard  
+    config = AugurConfig(session.logger, session)
+    os.environ['GITHUB_AUTH_TOKEN'] = config.get_section("Keys")['github_api_key']#self.config['gh_api_key']
+    
+    p= subprocess.run(['./scorecard', command, '--format=json'], cwd= path_to_scorecard ,capture_output=True, text=True, timeout=None)
+    session.logger.info('subprocess completed successfully... ')
+    output = p.stdout
+
     try:
-        generate_deps_data(session,repo_id, absolute_repo_path)
-    except Exception as e:
-        session.logger.error(f"Could not complete deps_model!\n Reason: {e} \n Traceback: {''.join(traceback.format_exception(None, e, e.__traceback__))}")
\ No newline at end of file
+        required_output = json.loads(output)
+    except json.decoder.JSONDecodeError as e:
+        session.logger.error(f"Could not parse required output! \n output: {output} \n Error: {e}")
+        return
+
+    session.logger.info('adding to database...')
+    session.logger.debug(f"output: {required_output}")
+
+    if not required_output['checks']:
+        session.logger.info('No scorecard checks found!')
+        return
+    
+    #Store the overall score first
+    overall_deps_scorecard = {
+        'repo_id': repo_id,
+        'name': 'OSSF_SCORECARD_AGGREGATE_SCORE',
+        'scorecard_check_details': required_output['repo'],
+        'score': required_output['score'],
+        'tool_source': 'scorecard_model',
+        'tool_version': '0.43.9',
+        'data_source': 'Git',
+        'data_collection_date': datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
+    }
+
+    session.insert_data(overall_deps_scorecard, RepoDepsScorecard, ["repo_id","name"])
+
+    #Store misc data from scorecard in json field. 
+    for check in required_output['checks']:
+        repo_deps_scorecard = {
+            'repo_id': repo_id,
+            'name': check['name'],
+            'scorecard_check_details': check,
+            'score': check['score'],
+            'tool_source': 'scorecard_model',
+            'tool_version': '0.43.9',
+            'data_source': 'Git',
+            'data_collection_date': datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
+        }
+        session.insert_data(repo_deps_scorecard, RepoDepsScorecard, ["repo_id","name"])
```

Comment:
@IsaacMilarky I think this should insert all the data at once so we aren't creating a database transaction for every piece of data

Reply:
agree